platform-core-module/src/main/java/com/softicar/platform/core/module/file/smb/ISmbEntry.java

```diff
@@ -0,0 +1,143 @@
+package com.softicar.platform.core.module.file.smb;
+
+import com.softicar.platform.common.core.exceptions.SofticarIOException;
+import com.softicar.platform.common.date.DayTime;
+import java.util.Optional;
+
+/**
+ * Represents a file or directory on an SMB share.
+ *
+ * @author Alexander Schmidt
+ */
+public interface ISmbEntry {
+
+	/**
+	 * Returns the name of this entry, inside its parent directory.
+	 * <p>
+	 * If this entry a directory, a the returned {@link String} will have a
```

Comment:
If this entry is a directory, the returned {@link String} will have a tailing slash.